sphinx/tutorial/rose/furthertopics/widget-development.rst

```diff
@@ -0,0 +1,156 @@
+Widget Development
+==================
+
+
+The ``rose edit`` GUI displays configurations using built-in widgets. For more
+complex requirements ``rose edit`` supports custom widgets as plugins.
+
+In this tutorial we will write a custom widget which offers typing suggestions
+when entering usernames.
+
+.. image:: img/rose-edit-custom-widget.png
+   :align: center
+   :width: 450px
+
+.. warning::
+
+   If you find yourself needing to write a custom widget, please contact the
+   Rose team for guidance.
+
+Example
+-------
+
+Create a new rose app by running the following command replacing
+``DIRECTORY`` with the path in which to create the suite::
+
+   rose tutorial widget DIRECTORY
+   cd DIRECTORY
+
+You will now have a rose app which contains the following files:
+
+.. code-block:: none
+
+   DIRECTORY
+   |-- meta
+   |   `-- lib
+   |       `-- python
+   |           `-- widget
+   |               |-- __init__.py
+   |               `-- username.py
+   `-- rose-app.conf
+
+The ``rose-app.conf`` file defines an environment variable called ``USER``:
+
+.. code-block:: rose
+
+   [env]
+   USER=fred
```

Comment:
It also (as I have loaded from your branch) includes a topmost section ```[command]``` defining a bash command ```default=echo "Hello ${USER}!"``` to run. I do not think this was meant to be there as it doesn't have any effect - I have left it in and then removed it as a test and the widget still works as disclosed in both cases. I imagine we will want to remove it.

Reply:
The command is meant to be there. Its so that the app can be run.